src/main/webapp/app/localci/build-agents/build-agent-details/build-agent-details/build-agent-details.component.ts

```diff
@@ -40,29 +63,74 @@ export class BuildAgentDetailsComponent implements OnInit, OnDestroy {
     private readonly route = inject(ActivatedRoute);
     private readonly buildQueueService = inject(BuildQueueService);
     private readonly alertService = inject(AlertService);
+    private readonly modalService = inject(NgbModal);
 
     protected readonly TriggeredByPushTo = TriggeredByPushTo;
     buildAgent: BuildAgentInformation;
+    buildJobStatistics: BuildJobStatistics = new BuildJobStatistics();
+    runningBuildJobs: BuildJob[] = [];
     agentName: string;
-    websocketSubscription: Subscription;
-    restSubscription: Subscription;
+    agentDetailsWebsocketSubscription: Subscription;
+    runningJobsWebsocketSubscription: Subscription;
+    runningJobsSubscription: Subscription;
+    agentDetailsSubscription: Subscription;
+    buildDurationInterval: ReturnType<typeof setInterval>;
     paramSub: Subscription;
     channel: string;
+    readonly runningBuildJobsChannel = '/topic/admin/running-jobs';
+
+    finishedBuildJobs: FinishedBuildJob[] = [];
 
     //icons
-    faCircleCheck = faCircleCheck;
-    faExclamationCircle = faExclamationCircle;
-    faExclamationTriangle = faExclamationTriangle;
-    faTimes = faTimes;
+    readonly faCircleCheck = faCircleCheck;
+    readonly faExclamationCircle = faExclamationCircle;
+    readonly faExclamationTriangle = faExclamationTriangle;
+    readonly faTimes = faTimes;
+    readonly faPauseCircle = faPauseCircle;
     readonly faPause = faPause;
     readonly faPlay = faPlay;
+    readonly faSort = faSort;
+    readonly faSync = faSync;
+
+    //Filter
+    searchSubscription: Subscription;
+    search = new Subject<void>();
+    isLoading = false;
+    searchTerm?: string = undefined;
+    finishedBuildJobFilter: FinishedBuildJobFilter;
+    faFilter = faFilter;
+
+    totalItems = 0;
+    itemsPerPage = ITEMS_PER_PAGE;
+    page = 1;
+    predicate = 'buildSubmissionDate';
+    ascending = false;
 
     ngOnInit() {
         this.paramSub = this.route.queryParams.subscribe((params) => {
             this.agentName = params['agentName'];
             this.channel = `/topic/admin/build-agent/${this.agentName}`;
```

Comment:
shall we also move the string to a class attribute? e.g., when replacing the REST prefixes, I found it particularly hard to find occurrences nested in components.